### ISSUE TITLE
Fireplace requires heartbeat signal

### ISSUE THREAD

**Issue by alkalinecoffee** [alkalinecoffee/echo_fireplace#1]
The fireplace includes a safety mechanism that requires the remote to sound out a heartbeat signal every 15 minutes.  I originally implemented this to re-send the ON signal every so often, but this does not appear to work.  Will need to capture signals from the remote and see if there is a different code for sending the heartbeat signal.

**Comment by robross0606** [comment]
Incidentally, I noticed something that looked a heckuva lot like a periodic heartbeat "blip" when I was capturing my remote signals in URH.  The length of the blip is extremely short -- much shorter than what I originally thought was my bit length (URH term).  I haven't gotten to the point where I can test my decoded signal yet, so I'm not sure if my bit length interpretation was even correct.  Were you able to solve this heartbeat problem?

**Comment by alkalinecoffee** [comment]
Nope, I have not had a chance to look at it.  The signal doesn't look very clean, so I'm not even sure it's sending a code rather than gibberish.  I'm not too familiar how this is implemented so it'll require a little more research.  Maybe the remote's FCC data can shed some light on it.

**Comment by robross0606** [comment]
Now this has me second-guessing my "bit length" which I really thought I had finally cracked on my signal.  I could see definite repeating patterns (preamble, gaps between repeats, length of signal, etc.) in the various signals I recorded.  But maybe my length value (482) is way too long and the actual binary signal is much longer.  Otherwise, like you, the heartbeat is "there" but *extremely* short.

My biggest fear is that the communications might bi-directional.  Does the remote have to receive any data back from the fireplace?  Does the [FCC data](https://fccid.io/K9L3001TX) tell us if there is any Rx capability in the remote?  They definitely call it a "Transmitter".

**Comment by robross0606** [comment]
Sitting down to tackle this today and immediately overwhelmed by my lack of experience with Arduino.  I'm a seasoned developer (Java, Android, etc.) and have plenty of experience hacking devices, but this is tera incognita for me.  I bought all the same parts you recommended (HiLetgo NodeMCU, etc.).  Any advice on how to get started with tooling?  Okay to use Windows or should I stick with a Linux toolchain?  Any good documentation or tutorials you can recommend for a complete Noob WRT Arduino?
